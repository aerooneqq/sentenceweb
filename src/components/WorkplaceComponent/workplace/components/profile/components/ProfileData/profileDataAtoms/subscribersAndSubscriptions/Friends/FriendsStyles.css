#friendsScrollViewer{ 
    width: 90%;
    height: 360px;

    overflow-y: auto;

    margin-left: 5%;
    margin-top: 40px;

    background: transparent;
}

#friendsScrollViewer::-webkit-scrollbar{ 
    width: 3px;
}

#friendsScrollViewer::-webkit-scrollbar-thumb{ 
    background: orangered;
    border-radius: 2px;
}