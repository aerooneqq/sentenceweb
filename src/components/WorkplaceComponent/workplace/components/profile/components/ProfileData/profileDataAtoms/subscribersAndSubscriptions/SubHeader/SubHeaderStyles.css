#subHeader{ 
    width: 100%;
    height: 30px;
    background:#f0f0f0;

    display: flex;
    flex-direction: row;

    margin-top:-20px;
}

.subHeaderOption{
    height: 100%; 
    flex-grow: 1;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.subHeaderOption:nth-child(1){ 
    border-style: solid;
    border-width: 0;
    border-right-width: 0.05px;
    border-color: gray;
}

.subHeaderOption:nth-child(2){ 
    border-style: solid;
    border-width: 0;
    border-left-width: 0.05px;
    border-color: gray;
}

.subHeaderOption:hover .subHeaderOptionHighlighter{ 
    background: orangered;
}

.sunHeaderOptionTitleClicked{ 
    color: orangered !important;
}

.subHeaderOptionTitle{
    flex-grow: 1;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    cursor: default;

    font-family: Arial;
    color: gray;
}

.subHeaderOptionHighlighter{ 
    width: 100%;
    height: 2px;
    background: transparent;

    border-style: none;
    border-radius:  2px;
}