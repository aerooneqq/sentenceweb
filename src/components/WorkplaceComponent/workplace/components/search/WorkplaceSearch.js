import React, {Component} from "react"

//Styles
import "./WorkplaceSearchStyles.css"

export default class WorkplaceSearch extends Component { 
    constructor(props) {
        super(props);

        this.state = { 
            value: null
        };

        this.handleInputValueChange = this.handleInputValueChange.bind(this);
        this.handleInputFocus = this.handleInputFocus.bind(this);
        this.handleSearchInputKeyEnter = this.handleSearchInputKeyEnter.bind(this);
    }

    handleInputValueChange(event) { 
        this.setState({ 
            value: event.target.value
        });
    }   

    componentDidMount(){ 
        let workplaceSearch = document.getElementById("workplaceSearchInput");

        workplaceSearch.addEventListener("focus", this.handleInputFocus);
        workplaceSearch.addEventListener("focusout", this.handleInputLostFocus);
    }

    handleInputLostFocus(){ 
        let workplaceSearchHelp = document.getElementById("workplaceSearchInputHelpCont"); 

        workplaceSearchHelp.classList.remove("collapsed");
        workplaceSearchHelp.classList.remove("fadeInAnimation");
        workplaceSearchHelp.classList.toggle("fadeOutAnimation");
    }

    handleInputFocus() {
        let workplaceSearchHelp = document.getElementById("workplaceSearchInputHelpCont"); 

        workplaceSearchHelp.classList.remove("collapsed");
        workplaceSearchHelp.classList.remove("fadeOutAnimation");
        workplaceSearchHelp.classList.toggle("fadeInAnimation");
    }

    handleSearchInputKeyEnter(target){ 
        if (target.charCode === 13){ 
            let searchInput = document.getElementById("workplaceSearchInput").value;
            if (this.props.search) { 
                this.props.search(searchInput);
            }
        }
    }

    render() {
        return(
            <div>
                <div id = "workplaceSearch">
                    <input type = "text"    
                           id = "workplaceSearchInput"
                           style = {{backgroundColor: this.props.backgroundColor === undefined ? "white" : this.props.backgroundColor,
                                     "margin-top": (this.props.marginTop ? this.props.marginTop : 0) + "px"}}
                           value = {this.state.value} 
                           onChange = {this.handleInputValueChange}
                           onKeyPress = {this.handleSearchInputKeyEnter}
                           placeholder = "Enter the query..." />
                </div>
                <div id = "workplaceSearchInputHelpCont"
                     className = "collapsed"
                     style = {{backgroundColor: this.props.backgroundColor === undefined ? "white" : this.props.backgroundColor}}>
                    Press enter
                </div>
            </div>
        );
    }
}