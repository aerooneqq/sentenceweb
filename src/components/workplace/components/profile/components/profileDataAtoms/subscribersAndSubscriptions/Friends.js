import React, {Component, lazy} from "react";

//Styles
import "./styles/FriendsStyles.css";

//Components
const SubHeader = lazy(() => import("./SubHeader"));
const FriendsSearch = lazy(() => import("./FriendsSearch"));
const Follower = lazy(() => import("./Follower"));

export default class Friends extends Component{ 
    constructor(props){ 
        super(props);
    }

    render(){ 
        return (
            <div id = "friendsContainer">
                <SubHeader />
                <FriendsSearch />
                <div id = "friendsScrollViewer">
                    <Follower />
                    <Follower />
                    <Follower />
                    <Follower />
                    <Follower />
                    <Follower />
                </div>
            </div>
        )
    }
}