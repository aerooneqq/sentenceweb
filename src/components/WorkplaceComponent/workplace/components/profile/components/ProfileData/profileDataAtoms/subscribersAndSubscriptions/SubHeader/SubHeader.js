import React, {Component} from "react";

//Styles
import "./SubHeaderStyles.css";

/**
 * PROPS LIST:
 * 1) changeUserFriendMode
 */
export default class SubsHeader extends Component{ 
    constructor(props){ 
        super(props);

        this.changeFriendModeToSubscribers = this.changeFriendModeToSubscribers.bind(this);
        this.changeFriendModeToSubscriptions = this.changeFriendModeToSubscriptions.bind(this);
        this.changeFrinedsMode = this.changeFriendsMode.bind(this);
    }

    changeFriendsMode(newStatus) { 
        this.props.changeUserFriendsMode(newStatus);
    }

    changeFriendModeToSubscribers() { 
        this.changeFrinedsState("subscribers");

        document.getElementById("subscribersHeader").classList
            .toggle("subHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .remove("subHeaderOptionTitleClicked");
    }

    changeFriendModeToSubscriptions() { 
        this.changeFrinedsState("subscriptions");

        document.getElementById("subscribersHeader").classList
            .remove("subHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .toggle("subHeaderOptionTitleClicked");
    }

    render(){ 
        return( 
            <div id = "subHeader">
                <div id = "subscribers" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscribers}>
                    <div id = "subscribersHeader" className = "subHeaderOptionTitle">Subscribers</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
                <div id = "subscriptions" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscriptions}>
                    <div id = "subscriptionsHeader" className = "subHeaderOptionTitle">Subscriptions</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
            </div>
        );
    }
}