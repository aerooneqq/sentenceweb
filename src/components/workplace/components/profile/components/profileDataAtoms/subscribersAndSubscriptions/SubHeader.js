import React, {Component} from "react";

//Styles
import "./styles/SubHeaderStyles.css";

export default class SubsHeader extends Component{ 
    constructor(props){ 
        super(props);

        this.changeFriendModeToSubscribers = this.changeFriendModeToSubscribers.bind(this);
        this.changeFriendModeToSubscriptions = this.changeFriendModeToSubscriptions.bind(this);
    }

    changeFriendModeToSubscribers() { 
        this.props.changeUserFriendMode("subscribers");

        document.getElementById("subscribersHeader").classList
            .toggle("sunHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .remove("sunHeaderOptionTitleClicked");
    }

    changeFriendModeToSubscriptions() { 
        this.props.changeUserFriendMode("subscriptions");

        document.getElementById("subscribersHeader").classList
            .remove("sunHeaderOptionTitleClicked");
        document.getElementById("subscriptionsHeader").classList
            .toggle("sunHeaderOptionTitleClicked");
    }

    render(){ 
        return( 
            <div id = "subHeader">
                <div id = "subscribers" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscribers}>
                    <div id = "subscribersHeader" className = "subHeaderOptionTitle">Subscribers</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
                <div id = "subscriptions" 
                     className = "subHeaderOption"
                     onClick = {this.changeFriendModeToSubscriptions}>
                    <div id = "subscriptionsHeader" className = "subHeaderOptionTitle">Subscriptions</div>
                    <div className = "subHeaderOptionHighlighter" />
                </div>
            </div>
        );
    }
}