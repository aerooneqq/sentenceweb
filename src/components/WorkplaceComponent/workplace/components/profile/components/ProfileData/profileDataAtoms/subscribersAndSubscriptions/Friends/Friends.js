import React, {Component, lazy} from "react";

//Styles
import "./FriendsStyles.css";
import Loader from "../../../../../../../../../loader/Loader";

//Services
import UserFriendsService from "../../../../../../../../../../services/UserServices/UserFriendsService";
import UserService from "../../../../../../../../../../services/UserServices/UserService";
import { alertAppMessage } from "../../../../../../../../../ApplicationMessage/ApplicationMessageManager";

//Components
import SubHeader from "../SubHeader/SubHeader";
import WorkplaceSearch from "../../../../../../search/WorkplaceSearch";
import Subscriber from "../Subscriber/Subscriber";
import Subscription from "../Subscription/Subscription";
import UserSearchResult from "../UserSearchResult/UserSearchResult";

export default class Friends extends Component {

    constructor(props){ 
        super(props);

        this.state = { 
            elements: [],
        };

        this.userFriendsService = new UserFriendsService(localStorage.getItem("token"));
        this.userService = new UserService(localStorage.getItem("token"));

        this.changeUserFriendMode = this.changeUserFriendMode.bind(this);
        this.deleteSubscriber = this.deleteSubscriber.bind(this);
        this.deleteSubscription = this.deleteSubscription.bind(this);
        this.uploadSubscribers = this.uploadSubscribers.bind(this);
        this.uploadSubscriptions = this.uploadSubscriptions.bind(this);
        this.searchForUsers = this.searchForUsers.bind(this);
        this.subscribe = this.subscribe.bind(this);
    }
    
    changeUserFriendMode(mode) { 
        if (mode === "subscribers"){ 
            this.uploadSubscribers();
        }
        else if (mode === "subscriptions"){ 
            this.uploadSubscriptions();
        }
    }

    deleteSubscriber(userID) {
        this.userFriendsService.deleteSubscriber(localStorage.getItem("token"), userID)
            .then(() => { 
                this.uploadSubscribers();
                alertAppMessage("The subscriber was deleted.", "success");
            }).catch(er =>{
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            })
    }

    deleteSubscription(userID) { 
        this.userFriendsService.deleteSubscription(userID)
            .then(() => { 
                this.uploadSubscriptions();
                alertAppMessage("The subscription was deleted.", "success");
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            })
    }

    subscribe(userID) { 
        this.userFriendsService.addSubscription(userID)
            .then(() => { 
                alertAppMessage("Added subscriber", "success");
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            })
    }

    uploadSubscribers() { 
        this.setState({ 
            elements: <Loader  message = "Loading subscribers..."/>
        });

        this.userFriendsService.getSubsribers()
            .then(res => { 
                let data = res.data;
                let components = [];

                for (let i = 0; i < data.length; i++){ 
                    components.push(<Subscriber subscriber = {data[i]}
                                                deleteSubscriber = {this.deleteSubscriber} />);
                }

                this.setState({
                    elements: components 
                });

                alertAppMessage("Subscribers were uploaded", "success");
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            });
    }

    uploadSubscriptions() { 
        this.setState({ 
            elements: <Loader message = "Loading subscriptions..." />
        });

        this.userFriendsService.getSubscriptions()
            .then(res => { 
                let data = res.data;
                let components = []; 

                for (let i = 0; i < data.length; i++){ 
                    components.push(<Subscription subscription = {data[i]}
                                                  deleteSubscription = {this.deleteSubscription} />)
                }

                this.setState({ 
                    elements: components,
                });

                alertAppMessage("Subscriptions were uploaded", "success");
            }).catch(er => { 
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            });
    }

    searchForUsers(login) { 
        this.setState({ 
            elements: <Loader message = "Searching for users..." />
        })

        this.userService.searchForUsers(login)
            .then(res => { 
                let data = res.data;

                let components = data.map(user => { 
                    return (
                        <UserSearchResult searchResult = {user} 
                                          subscribe = {this.subscribe}/>
                    )
                });

                this.setState({ 
                    elements: components
                });

                alertAppMessage("Search results were uploaded", "success");
            }).catch(er => {
                if (er.response) { 
                    alertAppMessage(er.response.data, "error");
                }
            });
    }

    render() {
        return (
            <div id = "friendsContainer">
                <SubHeader changeUserFriendMode = {this.changeUserFriendMode} />
                <WorkplaceSearch search = {this.searchForUsers}/>
                <div id = "friendsScrollViewer">
                    {this.state.elements}
                </div>
            </div>
        )
    }
}