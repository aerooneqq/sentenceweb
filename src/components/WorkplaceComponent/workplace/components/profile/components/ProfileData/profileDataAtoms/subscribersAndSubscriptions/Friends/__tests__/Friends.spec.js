import React from "react"; 
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import Friends from "../Friends";

import SubHeader from "../../SubHeader/SubHeader";
import WorkplaceSearch from "../../../../../../../search/WorkplaceSearch";

describe("Friends", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<Friends />))).toMatchSnapshot();
    });

    it("should have SubHeader and Workplace search", () => { 
        const friends = shallow(<Friends />);

        expect(friends.find(SubHeader).exists()).toBe(true);
        expect(friends.find(WorkplaceSearch).exists()).toBe(true);
    });

    it("should have proper initial state", () => { 
        const friends = shallow(<Friends />);

        const properState = { 
            elements: []
        }

        for (let key in properState) { 
            expect(properState[key]).toStrictEqual(friends.state(key));
        }
    });
});