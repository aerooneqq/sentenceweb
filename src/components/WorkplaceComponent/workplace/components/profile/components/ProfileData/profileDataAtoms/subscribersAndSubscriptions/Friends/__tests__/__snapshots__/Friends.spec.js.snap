// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends should render correctly 1`] = `
<div
  id="friendsContainer"
>
  <SubsHeader
    changeUserFriendMode={[Function]}
  />
  <WorkplaceSearch
    search={[Function]}
  />
  <div
    id="friendsScrollViewer"
  />
</div>
`;
